lib/boxcars/boxcar/active_record.rb

```diff
@@ -114,14 +115,52 @@ def safe_to_run?(code)
       true
     end
 
+    def change_count(changes_code)
+      return 0 unless changes_code
+
+      wrap_in_transaction do
+        # rubocop:disable Security/Eval
+        puts "computing change count with: #{changes_code}".colorize(:yellow)
+        eval changes_code
+        # rubocop:enable Security/Eval
+      end
+    end
+
+    def approved?(changes_code, code)
+      # find out how many changes there are
+      changes = change_count(changes_code)
+      puts "pending changes: #{changes} records".colorize(:yellow)
+      return true unless changes&.positive?
+
+      change_str = "#{changes} change#{'s' if changes.to_i > 1}"
+      raise SecurityError, "Can not run code that makes #{change_str} in read-only mode" if read_only?
+
+      return approval_function.call(changes, code) if approval_function.is_a?(Proc)
+
+      puts "This request will make #{change_str} changes. Are you sure you want to run it? (y/[n])"
+      answer = gets.chomp
+      answer.downcase == 'y'
+    end
+
+    # rubocop:disable Security/Eval
+    def run_active_record_code(code)
+      puts code.colorize(:yellow)
+      if read_only?
+        wrap_in_transaction do
+          eval code
```

Comment:
I think that right now is the perfect time to make your own wrapper method for `eval code` which will check for dangerous code inside.

like 
```
def evaluate_input(code)
   ... strip `code` from dangerous methods ...
  eval code
end
```

and use it all around, since you already started using it in many places and probably will be more

Reply:
good point.